sdk/secp256r1/src/lib.rs

```diff
@@ -0,0 +1,615 @@
+            };
+            assert_eq!(
+                test_case(1, &offsets),
+                Err(PrecompileError::InvalidDataOffsets)
+            );
+
+            let offsets = Secp256r1SignatureOffsets {
+                message_data_offset: 100,
+                message_data_size: 1000,
+                ..Secp256r1SignatureOffsets::default()
+            };
+            assert_eq!(
+                test_case(1, &offsets),
+                Err(PrecompileError::InvalidDataOffsets)
+            );
+
+            let offsets = Secp256r1SignatureOffsets {
+                message_data_offset: u16::MAX,
+                message_data_size: u16::MAX,
+                ..Secp256r1SignatureOffsets::default()
+            };
+            assert_eq!(
+                test_case(1, &offsets),
+                Err(PrecompileError::InvalidDataOffsets)
+            );
+        }
+
+        #[test]
+        fn test_pubkey_offset() {
+            let offsets = Secp256r1SignatureOffsets {
+                public_key_offset: u16::MAX,
+                ..Secp256r1SignatureOffsets::default()
+            };
+            assert_eq!(
+                test_case(1, &offsets),
+                Err(PrecompileError::InvalidDataOffsets)
+            );
+
+            let offsets = Secp256r1SignatureOffsets {
+                public_key_offset: 100 - (COMPRESSED_PUBKEY_SERIALIZED_SIZE as u16) + 1,
+                ..Secp256r1SignatureOffsets::default()
+            };
+            assert_eq!(
+                test_case(1, &offsets),
+                Err(PrecompileError::InvalidDataOffsets)
+            );
+        }
+
+        #[test]
+        fn test_signature_offset() {
+            let offsets = Secp256r1SignatureOffsets {
+                signature_offset: u16::MAX,
+                ..Secp256r1SignatureOffsets::default()
+            };
+            assert_eq!(
+                test_case(1, &offsets),
+                Err(PrecompileError::InvalidDataOffsets)
+            );
+
+            let offsets = Secp256r1SignatureOffsets {
+                signature_offset: 100 - (SIGNATURE_SERIALIZED_SIZE as u16) + 1,
+                ..Secp256r1SignatureOffsets::default()
+            };
+            assert_eq!(
+                test_case(1, &offsets),
+                Err(PrecompileError::InvalidDataOffsets)
+            );
+        }
+
+        #[test]
+        fn test_secp256r1() {
+            solana_logger::setup();
+            let message_arr = b"hello";
+            let instruction = new_secp256r1_instruction(message_arr).unwrap();
+            let mint_keypair = Keypair::new();
+            let feature_set = FeatureSet::all_enabled();
+
+            let tx = Transaction::new_signed_with_payer(
+                &[instruction.clone()],
+                Some(&mint_keypair.pubkey()),
+                &[&mint_keypair],
+                Hash::default(),
+            );
+
+            assert!(tx.verify_precompiles(&feature_set).is_ok());
+
```

Comment:
For completeness, can we add a test to check that the verification fails on a valid signature where the `s` value is in the upper half of the field?